js/client-mainmenu.js

```diff
@@ -114,7 +125,68 @@
 				);
 			}
 		},
+		createLoginFormObserver:function (self) {
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
+				if (window.app.user.get("named") && window.Storage.prefs("page") == "challenge") {
+					window.Storage.prefs("page", "testclient");
+					if ($('.challenge-container').length == 0) {
+						//user log in event trigerd when user log in and make user pick a team
+						$('body').on("login", function () {
+							self.onUserLogIn(self);
+						});
+						$('body').trigger("login");
```

Comment:
Why do you add a listener and then immediately trigger it? Why not just call the function directly?